src/libs/actions/Policy.js

```diff
@@ -682,6 +683,44 @@ function hasPolicyMemberError(policyMemberList) {
     return _.some(policyMemberList, member => !_.isEmpty(member.errors));
 }
 
+/**
+ * @param {String} value
+ * @returns {String}
+ */
+function capitalizeFirstLetter(value) {
+    return value.charAt(0).toUpperCase() + value.slice(1);
+}
+
+/**
+ * Generate a policy name based on an email and policy list.
+ * @param {String} email
+ * @param {Array} policyList
+ * @returns {String}
+ */
+function generateDefaultWorkspaceName(email, policyList) {
+    const policyListLength = policyList.length;
+    const hasPolicies = policyListLength > 0;
+    const emailParts = email.split('@');
+    let defaultWorkspaceName = '';
+    if (!emailParts || emailParts.length !== 2) {
+        return defaultWorkspaceName;
+    }
+    const username = emailParts[0];
+    const domain = emailParts[1];
+
+    if (_.includes(PUBLIC_DOMAINS, domain.toLowerCase())) {
+        defaultWorkspaceName = `${capitalizeFirstLetter(username)}'s Workspace`;
+    } else {
+        defaultWorkspaceName = `${capitalizeFirstLetter(domain.split('.')[0])}'s Workspace`;
+    }
+
+    if (`@${domain.toLowerCase()}` === CONST.SMS.DOMAIN) {
+        defaultWorkspaceName = 'My Group Workspace';
+    }
+
+    return hasPolicies ? `${defaultWorkspaceName} ${policyListLength + 1}` : defaultWorkspaceName;
```

Comment:
Close, this isn't necessarily what we want because if we have 3 workspaces that are named different things, we will have

Workspace Bla
Workspace BlaBla
My Group Workspace 3

The idea is that it should look like this

Workspace Bla
Workspace BlaBla
My Group Workspace

And then if we add another workspace that has the same name, it would be like this

Workspace Bla
Workspace BlaBla
My Group Workspace
My Group Workspace 1

Does that make sense?

Reply:
That makes a lot of sense, looking into it now